perfkitbenchmarker/providers/alicloud/ali_virtual_machine.py

```diff
@@ -289,6 +284,12 @@ def _Create(self):
           '--InternetMaxBandwidthIn %s' % self.bandwidth_in,
           '--InternetMaxBandwidthOut %s' % self.bandwidth_out])
 
+    # Create user and add SSH key
+    public_key = AliCloudKeyFileManager.GetPublicKey()
+    user_data = util.ADD_USER_TEMPLATE.format(self.user_name, public_key)
+    logging.debug('encoding startup script: %s' % user_data)
+    create_cmd.extend(['--UserData', base64.b64encode(user_data)])
```

Comment:
Have you tested this in Python2 and Python3?
b64encode returns bytes in Python3.  See https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/commit/bd5b51cd6daa5314a0eee0fbe92a125dc0dc2fe7 for a potential workaround.

Reply:
Thanks for the suggestion, I have implemented a fix and tested on python2 and python3.